hazelcast-spring/src/main/resources/hazelcast-spring-4.1.xsd

```diff
@@ -3836,6 +3838,47 @@
             <xs:enumeration value="POOLED"/>
         </xs:restriction>
     </xs:simpleType>
+    <xs:complexType name="persistent-memory">
+        <xs:annotation>
+            <xs:documentation>
+                Configuration for persistent memory (e.g. Intel Optane) devices.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:all>
+            <xs:element name="directories">
+                <xs:annotation>
+                    <xs:documentation>
+                        List of directories where the persistent memory
+                        is mounted to.
+                    </xs:documentation>
+                </xs:annotation>
+                <xs:complexType>
+                    <xs:choice maxOccurs="unbounded">
+                        <xs:element name="directory" type="persistent-memory-directory"/>
+                    </xs:choice>
+                </xs:complexType>
+            </xs:element>
+        </xs:all>
+    </xs:complexType>
+    <xs:complexType name="persistent-memory-directory">
+        <xs:annotation>
+            <xs:documentation>
+                The directory where persistent memory is mounted to.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:simpleContent>
+            <xs:extension base="xs:string">
```

Comment:
I guess you can also use something like `parameterizedValueType` as a base or some other type with restrictions.

Reply:
Not sure about this one. Do you have something concrete restriction in mind? All I can think of is restricting to `^(\/[\w_-]+)+(\/)?$` pattern, but I'm not sure if it makes sense here. I see a bigger risk in a possibly wrong pattern then leaving it unrestricted.

Reply:
Nothing in particular, just wanted to see if you had something in mind, like it not being empty or something like that. Not really a big concern.